src/Utils/Hexagons.svelte

```diff
@@ -148,64 +171,178 @@ $: drawableHexagons = hexagons.map(({id, row, col}) => {
     if(e.stopPropagation) e.stopPropagation();
     if(e.preventDefault) e.preventDefault();
     mouseDown = true;
+    clearInterval(resendCommand);
+    clearTimeout(pendingTimeout);
+    try {
+        if (e.targetTouches.length > 0) {
+            isTouch = true;
+            numTouches = e.targetTouches.length;
+        }
+    } catch(error) { 
+        numTouches = 1;
+    }
+
+    if (isTouch) {
+        Xstart = e.targetTouches[0].clientX - boundRect.left;
+        Ystart = e.targetTouches[0].clientY - boundRect.top;
+    } else {
+        Xstart = e.offsetX;
+        Ystart = e.offsetY;
+    }
+
+    if(!isPreset) {
+        findActiveHexagons(e);
+        resendCommand = setInterval(findActiveHexagons, 500, e);
+    }
 }
+
 function handleTouchMove(e) {
     if(e.stopPropagation) e.stopPropagation();
     if(e.preventDefault) e.preventDefault();
-
     if(!mouseDown) {
         return;
     }
-
-    const radius = Math.sqrt(3) * hexagonSideLength / 2;
-    const radiusSquared = radius * radius;
-
-    for(var i = 0; i < drawableHexagons.length; i++) {
-        const x = drawableHexagons[i].x;
-        const y = drawableHexagons[i].y;
-        const xDist = (x - e.offsetX);
-        const yDist = (y - e.offsetY);
-        const euclidianDistSquared = xDist * xDist + yDist * yDist;
-        if(euclidianDistSquared < radiusSquared) {
-            activeHexagon = drawableHexagons[i].id;
-            buildCommandBlocks(activeHexagon);
-            (async () => {
-                return await Com.hitEndpoint(endpoint, nopRoute, $act_command);
-            })().then(result => {
-                if(pendingTimeout) {
-                    clearTimeout(pendingTimeout);
-                }
-                pendingTimeout = setTimeout(() => { activeHexagon = -1 }, 500);
-                return result;
-            }).catch(error => {
-                message = error;
-                activeHexagon = -1;
-                throw error;
-            });
+    clearInterval(resendCommand);
+    if(!isPreset) {
+        findActiveHexagons(e);
+        if (pendingTimeout) { 
+            clearTimeout(pendingTimeout);
         }
+        pendingTimeout = setTimeout(() => {resendCommand = setInterval(findActiveHexagons, 500, e)}, 500);
+    }
+    if (isTouch) {
+        Xend = e.targetTouches[0].clientX - boundRect.left;
+        Yend = e.targetTouches[0].clientY - boundRect.top;
+    } else {
+        Xend = e.offsetX;
+        Yend = e.offsetY;
     }
 }
+
 function handleTouchEnd(e) {
     if(e.stopPropagation) e.stopPropagation();
     if(e.preventDefault) e.preventDefault();
-
+    clearInterval(resendCommand);
     mouseDown = false;
-    activeHexagon = -1;
+    isTouch = false;
+    activeHexagon = [];
+    if (isPreset) {
+        let temp = $OP_Mode;
+        if (presetName == "sweep") {
+            if (deltaX > 0 && Math.abs(deltaY) < 75 ){
+                    console.log('LR');
+                    $OP_Mode = parseInt("0x86",16);
```

Comment:
As a rule of thumb, do not use strings to represent integers, unless handling providing user input. This applies to all places using strings when manipulating integers. It can significantly hinder performance, hide bugs, and add surprising code paths.
```suggestion
                    $OP_Mode = 0x86
```